File: platform/backend/AiDevRequest.API/Entities/OrgMemory.cs
namespace AiDevRequest.API.Entities;

public class OrgMemory
{
    public Guid Id { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string Scope { get; set; } = "user"; // user, org
    public string Category { get; set; } = string.Empty; // preference, decision, pattern, standard, runbook
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public string SourceProject { get; set; } = string.Empty;
    public double Relevance { get; set; }
    public int UsageCount { get; set; }
    public string TagsJson { get; set; } = "[]"; // JSON array of tags
    public string EmbeddingStatus { get; set; } = "pending"; // pending, indexed, failed
    public int? DevRequestId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: platform/frontend/src/api/orgmemory.ts
import { getAuthHeaders } from './auth'

const API = import.meta.env.VITE_API_URL || ''

export interface OrgMemory {
  id: string
  userId: string
  scope: string
  category: string
  title: string
  content: string
  sourceProject: string
  relevance: number
  usageCount: number
  tagsJson: string
  embeddingStatus: string
  devRequestId: number | null
  createdAt: string
  updatedAt: string
}

export interface MemoryCategory {
  id: string
  name: string
  description: string
  color: string
}

export interface OrgMemoryStats {
  totalMemories: number
  userMemories: number
  orgMemories: number
  indexedCount: number
  avgRelevance: number
  totalUsages: number
  byCategory: { category: string; count: number; avgRelevance: number }[]
  byScope: { scope: string; count: number }[]
}

export async function listMemories(scope?: string, category?: string): Promise<OrgMemory[]> {
  const params = new URLSearchParams()
  if (scope) params.set('scope', scope)
  if (category) params.set('category', category)
  const res = await fetch(`${API}/api/org-memory/memories?${params}`, { headers: getAuthHeaders() })
  if (!res.ok) return []
  return res.json()
}

export async function createMemory(req: {
  title?: string
  content?: string
  scope?: string
  category?: string
  sourceProject?: string
  tags?: string[]
}): Promise<OrgMemory> {
  const res = await fetch(`${API}/api/org-memory/memories`, {
    method: 'POST',
    headers: { ...getAuthHeaders(), 'Content-Type': 'application/json' },
    body: JSON.stringify(req),
  })
  return res.json()
}

export async function deleteMemory(id: string): Promise<void> {
  await fetch(`${API}/api/org-memory/memories/${id}`, {
    method: 'DELETE',
    headers: getAuthHeaders(),
  })
}

export async function searchMemories(query: string, scope?: string): Promise<OrgMemory[]> {
  const res = await fetch(`${API}/api/org-memory/search`, {
    method: 'POST',
    headers: { ...getAuthHeaders(), 'Content-Type': 'application/json' },
    body: JSON.stringify({ query, scope }),
  })
  if (!res.ok) return []
  return res.json()
}

export async function getMemoryStats(): Promise<OrgMemoryStats> {
  const res = await fetch(`${API}/api/org-memory/stats`, { headers: getAuthHeaders() })
  return res.json()
}

export async function getCategories(): Promise<MemoryCategory[]> {
  const res = await fetch(`${API}/api/org-memory/categories`)
  if (!res.ok) return []
  return res.json()
}
